### ISSUE TITLE
Explanation of training results needed. 

### ISSUE THREAD

**Issue by shuait** [my89/SituationCrf#4]
Finally got the demo code works, and trained for 504 epochs. Could you explain what these results say? Especially the top1 and top5 acc.

 I1130 01:03:41.247375 25357 structure_frame_accuracy_layer.cpp:133] clearing...
I1130 01:03:41.248628 25357 structure_frame_accuracy_layer.cpp:133] clearing...
I1130 01:03:41.248723 25357 caffe.cpp:308] Batch 503, crf-log-loss = -6.99242
I1130 01:03:41.248740 25357 caffe.cpp:308] Batch 503, top1-*any-acc = 0.04
I1130 01:03:41.248752 25357 caffe.cpp:308] Batch 503, top1-*arg-acc = 0.44
I1130 01:03:41.248764 25357 caffe.cpp:308] Batch 503, top1-*full-acc = 0
I1130 01:03:41.248775 25357 caffe.cpp:308] Batch 503, top1-any-acc = 0
I1130 01:03:41.248782 25357 caffe.cpp:308] Batch 503, top1-arg-acc = 0
I1130 01:03:41.248790 25357 caffe.cpp:308] Batch 503, top1-full-acc = 0
I1130 01:03:41.248796 25357 caffe.cpp:308] Batch 503, top1-verb-acc = 0
I1130 01:03:41.248811 25357 caffe.cpp:308] Batch 503, top5-*any-acc = 0.04
I1130 01:03:41.248819 25357 caffe.cpp:308] Batch 503, top5-*arg-acc = 0.44
I1130 01:03:41.248826 25357 caffe.cpp:308] Batch 503, top5-*full-acc = 0
I1130 01:03:41.248833 25357 caffe.cpp:308] Batch 503, top5-any-acc = 0
I1130 01:03:41.248842 25357 caffe.cpp:308] Batch 503, top5-arg-acc = 0
I1130 01:03:41.248849 25357 caffe.cpp:308] Batch 503, top5-full-acc = 0
I1130 01:03:41.248857 25357 caffe.cpp:308] Batch 503, top5-verb-acc = 0
I1130 01:03:41.248864 25357 caffe.cpp:313] Loss: -9.36972
I1130 01:03:41.248904 25357 caffe.cpp:325] crf-log-loss = -9.36972 (* 1 = -9.36972 loss)
I1130 01:03:41.248922 25357 caffe.cpp:325] top1-*any-acc = 0.0554366
I1130 01:03:41.248930 25357 caffe.cpp:325] top1-*arg-acc = 0.325088
I1130 01:03:41.248939 25357 caffe.cpp:325] top1-*full-acc = 0.043135
I1130 01:03:41.248946 25357 caffe.cpp:325] top1-any-acc = 0
I1130 01:03:41.248953 25357 caffe.cpp:325] top1-arg-acc = 0.000357143
I1130 01:03:41.248961 25357 caffe.cpp:325] top1-full-acc = 0
I1130 01:03:41.248968 25357 caffe.cpp:325] top1-verb-acc = 0.00198413
I1130 01:03:41.248975 25357 caffe.cpp:325] top5-*any-acc = 0.0554366
I1130 01:03:41.248985 25357 caffe.cpp:325] top5-*arg-acc = 0.325088
I1130 01:03:41.248993 25357 caffe.cpp:325] top5-*full-acc = 0.043135
I1130 01:03:41.249001 25357 caffe.cpp:325] top5-any-acc = 0
I1130 01:03:41.249008 25357 caffe.cpp:325] top5-arg-acc = 0.000357143
I1130 01:03:41.249017 25357 caffe.cpp:325] top5-full-acc = 0
I1130 01:03:41.249023 25357 caffe.cpp:325] top5-verb-acc = 0.00198413

**Comment by my89** [comment]
The code you ran does not train, it tests the model for 504 batches (each 50 images) on the dev set. It reports the dev performance of the model. 

I've modified the names in the network.proto to match more closely to the paper. The output now looks like:

crf-log-loss = -8.9045
top1-*gtverb-value = 0.67
top1-*gtverb-value-any = 0.28
top1-*gtverb-value-full = 0.1
top1-value = 0.345
top1-value-any = 0.18
top1-value-full = 0.1
top1-verb = 0.46
top5-*gtverb-value = 0.67
top5-*gtverb-value-any = 0.28
top5-*gtverb-value-full = 0.1
top5-value = 0.455
top5-value-any = 0.22
top5-value-full = 0.1
top5-verb = 0.62

CRF loss is the loss in Equation (4) of the paper. The other measures, as defined in the paper, are the following :

(1) verb - is the verb correct
(2) value - # (verb, role, noun) that match any of the references divided by total (verb, role, noun) in a single reference
(3) value -any - do all (verb, role, noun) match any (potentially different) references
(4) value -all - do all (verb, role, noun) match exactly one reference.

Each of these is reported at top-1, meaning the first prediction and top-5, meaning any prediction in the top-5. The CRF does not output the top-K scoring situations, but instead outputs the top scoring situations for each verb, ordered by total situation score. This makes sense because the value measures give zero points if the prediction does not match on verb.

Finally, we also report the value measures given ground truth verbs. This is the *gtverb lines. Its output twice somewhat pointlessly because the top-k layer always outputs it.
